pkg/rand/safe_rand_test.go

```diff
@@ -37,16 +37,14 @@ func TestIntn(t *testing.T) {
 }
 
 func TestShuffle(t *testing.T) {
-	r0 := NewSafeRand(time.Now().UnixNano())
-
 	s1 := []string{"1", "2", "3", "4", "5"}
 	s2 := make([]string, len(s1))
 	copy(s2, s1)
 
 	var same int
-	for retry := 0; retry < 1; retry++ {
+	for retry := 0; retry < 10; retry++ {
 		same = 0
-
+		r0 := NewSafeRand(time.Now().UnixNano())
```

Comment:
I agree we should try more than once (this `retry < 1` looks like a mistake in the first place), but I don't think it should be necessary to recreate a `NewSafeRand()` each time? Have you tried by simply setting the number of retries to `10`, without moving this line?

Reply:
I have verified that, in the case of the same seed, each call to r0.Shuffle will get a different array sequence;
Your suggestion is feasible, I will update this patch. Thanks!